### DISCUSSION TITLE
Support for private repositories

### DISCUSSION THREAD

**Discussion by varunsh-coder** [step-security/harden-runner#74] [Ideas]
As of now, `harden-runner` works for public repositories that use GitHub Action Workflows. The insights and recommended policies are open to the public. 

This is obviously not the right model for private GitHub repositories. For a private repository, the insights should only be visible to members of the repository or the organization, and one would need to login using their GitHub account to view the insights. 

To work for private repositories, one would need to install a GitHub App with `actions: read` permissions.

Please share ideas on how you would like the experience to be for private repositories.

**Comment by wallies** [comment]
Would love to pilot this out. Im fine with installing a github app that gives read actions permissions.
Would also like to make it configurable, after I set mode to block, to not send any insights anymore if possible

**Reply by varunsh-coder** [reply]
> Would love to pilot this out. Im fine with installing a github app that gives read actions permissions.
> Would also like to make it configurable, after I set mode to block, to not send any insights anymore if possible

Great to hear @wallies! Let me get this ready for pilot and get back to you. Thanks!

**Reply by Developer64688** [reply]
🙂

**Reply by varunsh-coder** [reply]
Hi @wallies, you can use it on private repositories now. Please install the [Harden Runner App](https://github.com/marketplace/harden-runner-app) and then insights for new workflow runs should be visible in the insights website. Please let me know if you face any issues or have any feedback. 

The configuration to not send insights in block mode is not done yet. It is being tracked [here](https://github.com/step-security/harden-runner/issues/89). It should be done in a week or two...

**Reply by varunsh-coder** [reply]
> Hi @wallies, you can use it on private repositories now. Please install the [Harden Runner App](https://github.com/apps/harden-runner-app) and then insights for new workflow runs should be visible in the insights website. Please let me know if you face any issues or have any feedback.
> 
> The configuration to not send insights in block mode is not done yet. It is being tracked [here](https://github.com/step-security/harden-runner/issues/89). It should be done in a week or two...

Hi @wallies the work item to not send telemetry is done, the documentation and insights website are also updated. You will need to use the latest version of the `harden-runner`. Best to use the insights website to copy the snippet. Please try it out and let me know if you have feedback. Thanks!

**Comment by varunsh-coder** [comment]
[Harden Runner App](https://github.com/marketplace/harden-runner-app), to use Harden-Runner Action in private repos, is now in the marketplace!
